impeller/entity/entity_pass.cc

```diff
@@ -931,6 +933,20 @@ bool EntityPass::OnRender(
         // for blending (otherwise the blend pass will end up executing before
         // all the previous commands in the active pass).
 
+        // If all previous elements were skipped due to clear color optimization
```

Comment:
If we applied the clear color, that means we have a single color/rectangle as a dst input. Rather than rendering the element and allocating the render target to do this, we could instead create a 1x1 host uploaded texture with the color already set - that would be much faster.

The ColorFilterContents::MakeBlend also has the option of specifying a "foreground" color, which I _believe_ applies to the destination - the same place you'd use this texture. If this works all you need to do is conditionally pass the clear color instead of another texture which should be much simpler.

Reply:
Changed this to pass the clear color as the foreground color for `ColorFilterContents::MakeBlend`